sample-app/src/components/InputDropdown.tsx

```diff
@@ -52,74 +47,120 @@ export default function InputDropdown({
   placeholder,
   screenReaderInstructions,
   screenReaderInstructionsId,
-  options,
-  optionIdPrefix,
+  screenReaderText,
+  children,
   onSubmit = () => {},
-  updateInputValue,
-  updateDropdown,
+  onInputChange,
+  onInputFocus,
   renderButtons = () => null,
   cssClasses
-}: Props): JSX.Element | null {
+}: React.PropsWithChildren<Props>): JSX.Element | null {
   const [{
-    focusedOptionIndex,
+    focusedSectionIndex,
     shouldDisplayDropdown,
   }, dispatch] = useReducer(reducer, {
-    focusedOptionIndex: undefined,
+    focusedSectionIndex: undefined,
     shouldDisplayDropdown: false,
   });
-  const focusOptionId = focusedOptionIndex === undefined
-    ? undefined
-    : `${optionIdPrefix}-${focusedOptionIndex}`;
 
+  const [focusedOptionId, setFocusedOptionId] = useState<string | undefined>(undefined);
   const [latestUserInput, setLatestUserInput] = useState(inputValue);
   const [screenReaderKey, setScreenReaderKey] = useState(0);
 
   const inputRef = useRef<HTMLInputElement>(document.createElement('input'));
+  const dropdownRef = useRef<HTMLDivElement>(document.createElement('div'));
 
   if (!shouldDisplayDropdown && screenReaderKey) {
     setScreenReaderKey(0);
   }
 
+  const childrenArray = React.Children.toArray(children);
+  const childrenWithProps = childrenArray.map((child, index) => {
+    if (!React.isValidElement(child)) {
```

Comment:
Should probably check if this child is a Section component. Otherwise child.props.onClickOption below would be undefined and would throw an error when called